libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3720,6 +3800,64 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 	});
 }
 
+string YulUtilFunctions::inlineArrayConversionFunction(InlineArrayType const& _from, ArrayType const& _to)
+{
+	if (_to.dataStoredIn(DataLocation::CallData))
+		solAssert(false);
+
+	if (!_to.isDynamicallySized())
+		solAssert(_to.length() == _from.components().size());
+
+	string functionName =
+		"convert_inline_array_" +
+		_from.identifier() +
+		"_to_" +
+		_to.identifier();
+
+	vector<map<string, string>> memberSetValues;
+	unsigned stackItemIndex = 0;
+
+	for (auto&& [index, type]: _from.components() | ranges::views::enumerate)
```

Comment:
```suggestion
	for (auto&& [index, type]: _from.components() | ranges::views::enumerate | ranges::views::reverse)
```

This should also work just fine, shouldn't it? That is if you also reverse ``stackItemIndex``...

I'm wondering, since I just had a look at the optimized IR code generated for cases along the lines of https://github.com/ethereum/solidity/pull/12883#discussion_r894315543...
and for something like
```
contract c {
    struct Point { uint256 x; uint256 y; }
    function test() public {
	Point[100] memory a = [
		Point(0,0), Point(0,0), ..., Point(0,0)
	];
    }
}
```
IR code generation actually works, but since it cannot reorder the allocations for all the ``Point(0,0)``s, it has to move an enormous amount of local variables to memory to be able to cope...
And it's a bit silly to move the inline array elements to memory variables, only to load them again from there and store them to different memory locations, so it'd be good, if we could avoid that :-).

The problem is, though, that even if we reverse this here, the evm code transform still has a hard time handling the resulting yul code optimally, I'm not yet quite sure what to do about that... but reversing this here would at least give it a chance to work better :-).